test/GnosisSafeMultisig.js

```diff
@@ -132,5 +132,63 @@ describe("GnosisSafeMultisig", function () {
+            await safeGuard.deployed();
+
+            const to = safeGuard.address;
+            const fallbackHandler = AddressZero;
+            const paymentToken = AddressZero;
+            const paymentReceiver = AddressZero;
+            const payment = 0;
+            let nonce = 0;
+            const payload = safeGuard.interface.encodeFunctionData("setGuardForSafe", [safeGuard.address]);
+            // Pack the data
+            const data = ethers.utils.solidityPack(["address", "address", "address", "address", "uint256", "uint256", "bytes"],
+                [to, fallbackHandler, paymentToken, paymentReceiver, payment, nonce, payload]);
+
+            // Create a multisig with the packed data
+            const tx = await gnosisSafeMultisig.create(defaultOwnerAddresses, threshold, data);
+            const result = await tx.wait();
+
+            // Check that the obtained address is not zero
+            expect(result.events[0].address).to.not.equal(AddressZero);
+            const proxyAddress = result.events[0].address;
+
+            // Get the safe multisig instance
+            const multisig = await ethers.getContractAt("GnosisSafe", proxyAddress);
+
+            // Check the multisig owners and threshold
+            const owners = await multisig.getOwners();
+            for (let i = 0; i < defaultOwnerAddresses.length; i++) {
+                expect(owners[i]).to.equal(defaultOwnerAddresses[i]);
+            }
+            expect(await multisig.getThreshold()).to.equal(threshold);
+
+            // Check the guard slot
+            const guardSlot = "0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8";
+            let slotValue = await ethers.provider.getStorageAt(multisig.address, guardSlot);
+            slotValue = ethers.utils.hexlify(ethers.utils.stripZeros(slotValue));
+            const safeGuardAddress = safeGuard.address.toLowerCase();
+            expect(slotValue).to.equal(safeGuardAddress);
+
+            // Send funds to the multisig address
+            await signers[0].sendTransaction({to: multisig.address, value: ethers.utils.parseEther("1000")});
+
+            // Try to send ETH funds from the multisig using first two Safe owners while facing the guard check
+            const safeContracts = require("@gnosis.pm/safe-contracts");
+            nonce = await multisig.nonce();
+            // The function call is irrelevant, we just need to pass the value
+            let txHashData = await safeContracts.buildContractCall(multisig, "nonce", [], nonce, 0, 0);
+            txHashData.value = ethers.utils.parseEther("10");
+            const signMessageData = [await safeContracts.safeSignMessage(signers[1], multisig, txHashData, 0),
+                await safeContracts.safeSignMessage(signers[2], multisig, txHashData, 0)];
+            await expect(
+                safeContracts.executeTx(multisig, txHashData, signMessageData, 0)
+            ).to.be.revertedWith("Guarded");
```

Comment:
This set of actions simulates the Safe instance call execution that tries to transfer some value, and it is blocked by the Safe Guard.